codex-ui/src/vue/components/alert/core/useAlert.ts

```diff
@@ -0,0 +1,109 @@
+import { ref } from 'vue';
+import type { AlertInterface, AlertOptions, AlertType } from './types';
+import { createSharedComposable } from '@vueuse/core';
+
+/**
+ * Store alert state
+ */
+const useStore = createSharedComposable(() => {
+  const alertStore = ref<AlertOptions[]>([]);
+
+  /**
+   * trigger alert component
+   * @param type - type of alert (success, error, warning, info and default)
+   * @param opt - alert options
+   */
+  function createAlertHandler(type: AlertType, opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>): void {
+    const alert = alertStore.value.findIndex((idx: AlertOptions) => idx.timeout === opt?.timeout);
+
+    alertStore.value.push({ type,
+      ...opt });
+
+    setTimeout(() => {
+      alertStore.value.splice(alert, alertStore.value.length);
+    }, opt?.timeout);
+  }
```

Comment:
what this method is need for? Description, name and code is not clear. What is "alert handler" ?